tests/test_equilibrium.py

```diff
@@ -375,3 +380,67 @@ def test_backward_compatible_load_and_resolve():
     f_obj = ForceBalance(eq=eq)
     obj = ObjectiveFunction(f_obj, use_jit=False)
     eq.solve(maxiter=1, objective=obj)
+
+
+@pytest.mark.unit
+def test_contract_equilibrium():
+    """Test contract_equilibrium utility function."""
+    eq = get("ESTELL")
+    rho = 0.5
+    eq_half_rho = contract_equilibrium(eq, rho)
+
+    # ensure the new surface is the same as the desired inner surface
+    surf_inner = eq.get_surface_at(rho)
+    np.testing.assert_allclose(surf_inner.R_lmn, eq_half_rho.surface.R_lmn, err_msg="R")
+    np.testing.assert_allclose(surf_inner.Z_lmn, eq_half_rho.surface.Z_lmn, err_msg="Z")
+    np.testing.assert_allclose(surf_inner.NFP, eq_half_rho.surface.NFP, err_msg="NFP")
+
+    # test geometry, profiles, |B| and |F| match btwn orig eq and new contracted eq
+    data_keys = ["|B|", "|F|", "R", "Z", "lambda", "p", "iota"]
+    contract_grid = LinearGrid(
+        rho=np.linspace(0, 1.0, eq.L),
+        M=eq.M_grid,
+        N=eq.N_grid,
+        NFP=eq.NFP,
+    )
+    grid = LinearGrid(
+        rho=np.linspace(0, rho, eq.L),
+        M=eq.M_grid,
+        N=eq.N_grid,
+        NFP=eq.NFP,
+    )
+    contract_data = eq_half_rho.compute(data_keys, grid=contract_grid)
+    data = eq.compute(data_keys, grid=grid)
+    for key in data_keys:
+        if key != "|F|":
+            np.testing.assert_allclose(contract_data[key], data[key], err_msg=key)
+    np.testing.assert_allclose(
+        contract_data["|F|"], data["|F|"], rtol=2e-4, err_msg="|F|"
+    )
+
+
+@pytest.mark.unit
+def test_scale_profile():
+    """Test scale_profile utility function."""
+    eq = get("SOLOVEV")
+    p = eq.pressure
+    p_spline = p.to_spline(method="cubic2")
+    p_zern = p.to_fourierzernike()
+    p_mtanh = p.to_mtanh(order=4, ftol=1e-12, xtol=1e-12)
```

Comment:
Missing `PowerSeriesProfile` and `TwoPowerProfile`. We should make this test do that parameterization thing where it tests on a list of classes

Reply:
`p` is already a powerseries profile